### PR TITLE
FIX: Privatize XdawnTransformer & SearchLight

### LINKED ISSUE #3592: Q: decoding module sufficiently stable for 0.13?
The next release is coming soon, and I am wondering whether the recent enh in the decoding module are sufficiently stable for it.

We've encountered very painful refactoring in the past because of lack of experience, and we should clearly avoid this mistake again.

I am most worried about :

1) the location of `XdawnTransformer` currently in mne.preprocessing, although may need to be moved to mne.decoding

2) the `SearchLight` and `GeneralizationLight`: although working & stable, these classes currently add little from the user's perspective, as they do, with a cleaner code, the same as `TimeDecoding` and `GeneralizationAcrossTime`. Also the name 'GeneralizationLight` may not be appropriate.

I reckon we should put them back to private until 0.14 to give us some distance. WDYT?

### PR DESCRIPTION
**Pull request by kingjr**
closes #3592

### CODE REVIEW

**Comment by kingjr** [comment]
ready for review

**Comment by Eric89GXL** [comment]
LGTM +1 for merge

**Comment by agramfort** [comment]
besides LGTM

**Comment by Eric89GXL** [comment]
@agramfort besides what, I don't see any comment

**Comment by kingjr** [comment]
I reverted back the GeneralizationAcrossTime and condition example to its previous form.

File: examples/decoding/plot_decoding_time_generalization_conditions.py [hunk 1]
```diff
@@ -5,11 +5,6 @@
 
 This example runs the analysis computed in:
 
-Jean-Remi King, Alexandre Gramfort, Aaron Schurger, Lionel Naccache
-and Stanislas Dehaene, "Two distinct dynamic modes subtend the detection of
-unexpected sounds", PLOS ONE, 2013,
-http://www.ncbi.nlm.nih.gov/pubmed/24475052
```
Comment (agramfort) [inline examples/decoding/plot_decoding_time_generalization_conditions.py]: not cool :(

**Comment by agramfort** [comment]
closing in favor of #3617

**Pull request closed without merging**